src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/WsTrust/WsTrustResponse.cs

```diff
@@ -148,9 +148,17 @@ internal static WsTrustResponse CreateFromResponseDocument(XDocument responseDoc
                     {
                         continue;
                     }
-                    
-                    tokenResponseDictionary.Add(tokenTypeElement.Value,
-                        requestedSecurityToken.FirstNode.ToString(SaveOptions.DisableFormatting));
+
+                    var token = new System.Text.StringBuilder();
+                    foreach (var node in requestedSecurityToken.Nodes())
+                    {
+                        // Since we moved from XDocument.Load(..., LoadOptions.None) to Load(..., LoadOptions.PreserveWhitespace),
+                        // requestedSecurityToken can contain multiple nodes, and the first node is possibly just whitespaces e.g. "\n   ",
+                        // so we concatenate all the sub-nodes to include everything
+                        token.Append(node.ToString(SaveOptions.DisableFormatting));
+                    }
+
+                    tokenResponseDictionary.Add(tokenTypeElement.Value, token.ToString());
```

Comment:
this is a dictionary? are we sure we won't have several tokens of the same type?